code/game/machinery/atmoalter/canister.dm

```diff
@@ -369,115 +320,103 @@ update_flag
 		ui.set_auto_update(1)
 
 
-/obj/machinery/portable_atmospherics/canister/ui_data(mob/user, datum/topic_state/state)
-	init_data_vars() //set up var/colorcontainer and var/possibledecals
-
-	// this is the data which will be sent to the ui
-	var/data[0]
-	data["name"] = name
-	data["menu"] = menu ? 1 : 0
-	data["canLabel"] = can_label ? 1 : 0
-	data["canister_color"] = canister_color
-	data["colorContainer"] = colorcontainer.Copy()
-	colorcontainer.Cut()
-	data["possibleDecals"] = possibledecals.Copy()
-	possibledecals.Cut()
+/obj/machinery/portable_atmospherics/canister/tgui_data()
+	init_data_vars() //set up var/colorcontainer
```

Comment:
Shouldnt this be in `Initialize()` instead of being called every single `tgui_data` call?

Reply:
refactored the whole init_data_vars() mess out of existence